lib/wasi/src/syscalls/legacy/snapshot0.rs

```diff
@@ -22,17 +23,18 @@ pub fn fd_filestat_get(
     let new_buf: WasmPtr<types::__wasi_filestat_t> = buf.cast();
 
     // Copy the data including the extra data
-    let new_filestat_setup: types::__wasi_filestat_t = wasi_try_mem!(new_buf.read(memory));
+    let new_filestat_setup: types::__wasi_filestat_t = wasi_try_mem!(new_buf.read(&ctx, memory));
```

Comment:
Does this compile? From what I read on the `WasmPtr` wasmer3 API the signature (https://github.com/wasmerio/wasmer/blob/wasmer3/lib/api/src/sys/ptr.rs#L186) only requires the memory reference.
Am I missing something?

Reply:
You are looking at the wrong branch: you need to look in https://github.com/wasmerio/wasmer/tree/context_api for the new API

Reply:
ahá! Thanks